package NSnippets.NProto;

option optimize_for = SPEED;

message TBounds {
    required int32 Width = 1;
    required int32 Height = 2;
    optional int32 Top = 3;
    optional int32 Left = 4;
    optional int32 Bottom = 5;
    optional int32 Right = 6;
}

message TUrlLink {
    required string Text = 1;
    required string Url = 2;
}

message TSerpItem {
    optional string TitleText = 1;
    optional string Url = 2;
    repeated TUrlLink UrlLinks = 3;
    repeated TUrlLink Sitelinks = 4;
    repeated string ItemText = 5;
    repeated string MetaText = 6;
    optional string InnerText = 7;
    optional TBounds Bounds = 8;
    optional bool IsAdvItem = 9;
}
